financial-connections/src/main/java/com/stripe/android/financialconnections/features/consent/ConsentViewModel.kt

```diff
@@ -52,33 +52,36 @@ internal class ConsentViewModel @Inject constructor(
                 shouldShowMerchantLogos = shouldShowMerchantLogos,
                 merchantLogos = sync.visual.merchantLogos
             )
-        }.execute { copy(consent = it) }
-    }
-
-    private fun logErrors() {
-        onAsync(
-            ConsentState::consent,
+        }.execute(
+            reducer = { copy(consent = it) },
             onSuccess = { eventTracker.track(PaneLoaded(Pane.CONSENT)) },
-            onFail = { logger.error("Error retrieving consent content", it) }
+            onFail = {
+                eventTracker.logError(
+                    extraMessage = "Error retrieving consent content",
+                    error = it,
+                    logger = logger,
+                    pane = Pane.CONSENT
+                )
+            },
         )
-        onAsync(ConsentState::acceptConsent, onFail = {
+    }
+
+    fun onContinueClick() = suspend {
```

Comment:
Nit: It feels odd to use `fun abc() = …` here. The method doesn’t return anything, but the syntax kinda makes it look like it. What do you think about sticking to the existing formatting?